app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt

```diff
@@ -365,6 +367,9 @@ class StateFragmentPresenter @Inject constructor(
   private fun subscribeToAnswerOutcome(
     answerOutcomeResultLiveData: LiveData<AsyncResult<AnswerOutcome>>
   ) {
+    if (stateViewModel.getCanSubmitAnswer().get() == true) {
+      recyclerViewAssembler.resetUserAnswerState()
+    }
```

Comment:
I don't understand why this is required. Could you please add a comment to explain it?

Reply:
That's redundant check. Removed.